src/packages/external-event/external-event.ts

```diff
@@ -0,0 +1,136 @@
+import type { Express, Request, Response } from 'express';
+import type { ExternalEventTypeInsertInput } from '../../types/external-event.js';
+import Ajv from 'ajv';
+import { getEnv } from '../../env.js';
+import getLogger from '../../logger.js';
+import gql from './gql.js';
+import { HasuraError } from '../../types/hasura.js';
+
+const logger = getLogger('packages/external-event/external-event');
+const { HASURA_API_URL } = getEnv();
+const GQL_API_URL = `${HASURA_API_URL}/v1/graphql`;
+
+const ajv = new Ajv();
+
+async function uploadExternalEventType(req: Request, res: Response) {
+  const authorizationHeader = req.get('authorization');
+
+  const {
+    headers: { 'x-hasura-role': roleHeader, 'x-hasura-user-id': userHeader },
+  } = req;
+
+  const { body } = req;
+  const { external_event_type_name, attribute_schema } = body;
+  logger.info(`POST /uploadExternalEventType: Uploading External Event Type: ${external_event_type_name}`);
+
+  const headers: HeadersInit = {
+    Authorization: authorizationHeader ?? '',
+    'Content-Type': 'application/json',
+    'x-hasura-role': roleHeader ? `${roleHeader}` : '',
+    'x-hasura-user-id': userHeader ? `${userHeader}` : '',
+  };
+
+  // Validate schema is valid JSON Schema
+  const schemaIsValid: boolean = ajv.validateSchema(attribute_schema);
+  if (!schemaIsValid) {
+    logger.error(
+      `POST /uploadExternalEventType: Schema validation failed for External Event Type ${external_event_type_name}`,
+    );
+    ajv.errors?.forEach(ajvError => logger.error(ajvError));
+    res.status(500).send({ message: ajv.errors });
+    return;
+  }
+
+  // Make sure name in schema (title) and provided name match
+  try {
+    if (attribute_schema['title'] === undefined || attribute_schema.title !== external_event_type_name) {
+      throw new Error('Schema title does not match provided external event type name.');
+    }
+  } catch (error) {
+    logger.error(
+      `POST /uploadExternalEventType: Error occurred during External Event Type ${external_event_type_name} upload`,
+    );
+    logger.error((error as Error).message);
+    res.status(500).send({ message: (error as Error).message });
+    return;
+  }
+
+  logger.info(`POST /uploadExternalEventType: Attribute schema was VALID`);
```

Comment:
Please use present tense here:
```suggestion
  logger.info(`POST /uploadExternalEventType: Attribute schema is VALID`);
```

Reply:
All `logger` and `response` messages now use present tense!